Add files via upload

KNaved opened the pull request:
Fix issue for Popout if URL has # tag in it as window.location.search will return null. This change will be fix issue for AngularJs Application having routing in it.

mattgodbolt commented:
Hi @KNaved  -- can you add the changes to the regular GoldenLayout files too; not just the distribution `goldenlayout.js`. Both `goldenlayout.js` and `goldenlayout.min.js` are autogenerated from the other files. GitHub isn't showing me what the diff is; I imagine it's quite small. If you have any trouble finding it, let me know and I'll try and work back from your change to the original files.

KNaved commented:
Hi @mattgodbolt ,

Are you able find changes now.  I have added new method **_lm.utils.getHashValue_** to get queryString from url having # in it. And modified **_lm.utils.getQueryStringParam_**

//Added by Me
lm.utils.getHashValue = function(key) {
var matches = location.hash.match(new RegExp(key+'=([^&]*)'));
return matches ? matches[1] : null;
};

lm.utils.getQueryStringParam = function (param) {
//--------Added by me
if(window.location.hash){
return lm.utils.getHashValue(param);
}
//----------------
else if (!window.location.search) {
return null;
}

var keyValuePairs = window.location.search.substr(1).split('&'),
params = {},
pair,
i;

for (i = 0; i < keyValuePairs.length; i++) {
pair = keyValuePairs[i].split('=');
params[pair[0]] = pair[1];
} return params[param] || null; };

mattgodbolt commented:
Hi @KNaved ; there's still some weird GitHub things going on: I'll update just the `DragSource.js` you've patched and I'll submit. Thank you so much for the fix.

mattgodbolt commented:
I merged your changes; but I think I lost the actual main fix. I'm now trying to find the diff and apply it to `utils.js` directly.
